omg/mechanics/animations.py

```diff
@@ -0,0 +1,275 @@
+        self.shoot_texture_index = 0
+        self.die_texture_index = 0
+
+        self.animation_timer = 0
+
+        # Key states
+        self.w_pressed = False
+        self.a_pressed = False
+        self.s_pressed = False
+        self.d_pressed = False
+
+        # Movement direction
+        self.move_direction = (0, 0)
+
+        # Action finish checker
+        self.action_finished = 0
+
+        self.player_change_x = 0
+        self.player_change_y = 0
+
+    def update(self, delta_time, player_change_x, player_change_y):
+        # Movement and game logic
+        self.player_change_x = player_change_x
+        self.player_change_y = player_change_y
+        self.update_animation(delta_time)
+
+        if self.player_state == WALK:
+            self.update_movement()
+
+        return self.player_texture
+    
+    def update_animation(self, delta_time):
+
+        # Figure out if we need to flip face left or right
+        if self.player_change_y < 0 and self.character_face_direction != DOWN_FACING:
+            self.character_face_direction = DOWN_FACING
+            self.idle_textures = self.idle_textures_down
+            self.walk_textures = self.walk_textures_down
+            self.slash_textures = self.slash_textures_down
+            self.spellcast_textures = self.spellcast_textures_down
+            self.thrust_textures = self.thrust_textures_down
+            self.shoot_textures = self.shoot_textures_down
+        if self.player_change_y > 0 and self.character_face_direction != UP_FACING:
+            self.character_face_direction = UP_FACING
+            self.idle_textures = self.idle_textures_up
+            self.walk_textures = self.walk_textures_up
+            self.slash_textures = self.slash_textures_up
+            self.spellcast_textures = self.spellcast_textures_up
+            self.thrust_textures = self.thrust_textures_up
+            self.shoot_textures = self.shoot_textures_up
+        if self.player_change_x < 0 and self.character_face_direction != LEFT_FACING:
+            self.character_face_direction = LEFT_FACING
+            self.idle_textures = self.idle_textures_left
+            self.walk_textures = self.walk_textures_left
+            self.slash_textures = self.slash_textures_left
+            self.spellcast_textures = self.spellcast_textures_left
+            self.thrust_textures = self.thrust_textures_left
+            self.shoot_textures = self.shoot_textures_left
+        if self.player_change_x > 0 and self.character_face_direction != RIGHT_FACING:
+            self.character_face_direction = RIGHT_FACING
+            self.idle_textures = self.idle_textures_right
+            self.walk_textures = self.walk_textures_right
+            self.slash_textures = self.slash_textures_right
+            self.spellcast_textures = self.spellcast_textures_right
+            self.thrust_textures = self.thrust_textures_right
+            self.shoot_textures = self.shoot_textures_right
```

Comment:
if we define that a `Texture` class mentioned in the constructor, we can also simplify the logic here. Let us discuss after we fixed the previous part.